src/us/kbase/workspace/test/kbase/ArgUtilsTest.java

```diff
@@ -150,5 +165,324 @@ private void failChooseInstant(
+								new ExternalDataUnit()
+										.withDataId("did")
+										.withDataUrl("https://data.com")
+										.withDescription("d")
+										.withResourceName("JGI")
+										.withResourceReleaseEpoch(70000L)
+										.withResourceUrl("https://otherdata.com")
+										.withResourceVersion("v2"),
+								new ExternalDataUnit().withDataId("d")
+								))
+						.withInputWsObjects(list("foo/bar", "1/1/1"))
+						.withIntermediateIncoming(list("a", "b"))
+						.withIntermediateOutgoing(list("c", "d"))
+						.withMethod("meth")
+						.withMethodParams(list(new UObject("foo"), new UObject(7L)))
+						.withScript("cooking_with_sudafed")
+						.withScriptCommandLine("--dump-in-bathtub --location=doublewide --teeth=7")
+						.withScriptVer("git er done")
+						.withService("methamphetaminedelivery")
+						.withServiceVer("some large number")
+						.withSubactions(list(
+								newSA()
+										.withCodeUrl("file://c:/stuff/methcode")
+										.withCommit("version control is for wussies")
+										.withEndpointUrl("https://uswest.methdeliv.amazon.com")
+										.withName("meth delivery sub service")
+										.withVer("what did I just say about version control"),
+								newSA().withVer("1")
+								)),
+				newPA() // test time strings vs epoch
+						.withTime("2021-04-23T12:01:01.456-0700")
+						.withExternalData(list(new ExternalDataUnit()
+								.withResourceReleaseDate("2020-03-22T11:00:00.345-0600")
+						)),
+				newPA().withCaller("c"), // test null external data list
+				// test empty lists 
+				newPA().withCaller("d").withExternalData(list()).withSubactions(list())
+				);
+		final Provenance expected = Provenance.getBuilder(U2, inst(20000))
+				.withAction(ProvenanceAction.getBuilder()
+						.withCaller("c")
+						.withCustom(ImmutableMap.of("foo", "bar"))
+						.withDescription("d")
+						.withTime(inst(90000))
+						.withExternalData(list(
+								ExternalData.getBuilder()
+										.withDataID("did")
+										.withDataURL("https://data.com")
+										.withDescription("d")
+										.withResourceName("JGI")
+										.withResourceReleaseDate(inst(70000L))
+										.withResourceURL("https://otherdata.com")
+										.withResourceVersion("v2")
+										.build(),
+								ExternalData.getBuilder().withDataID("d").build()
+								))
+						.withWorkspaceObjects(list("foo/bar", "1/1/1"))
+						.withIncomingArgs(list("a", "b"))
+						.withOutgoingArgs(list("c", "d"))
+						.withMethod("meth")
+						.withMethodParameters(list("foo", 7L))
+						.withScript("cooking_with_sudafed")
+						.withCommandLine("--dump-in-bathtub --location=doublewide --teeth=7")
+						.withScriptVersion("git er done")
+						.withServiceName("methamphetaminedelivery")
+						.withServiceVersion("some large number")
+						.withSubActions(list(
+								SubAction.getBuilder()
+										.withCodeURL("file://c:/stuff/methcode")
+										.withCommit("version control is for wussies")
```

Comment:
My understanding is that the plural of "wuss" is "wusses"; is "wussies" an Americanism? Or a portmanteau word combining "wusses" and "pussies"?

Reply:
aha! Maybe the "wussies" vs "wusses" distinction is regional then. Hmmm.